### DISCUSSION TITLE
Return the CLI after run some tasks

### DISCUSSION THREAD

**Discussion by jefpalma** [ktbyers/netmiko#2591] [Q&A]
I'm trying to write a script that connect to a router, run some tasks e after these tasks are finished the CLI of the router is delivered to me.

Is there any method of netmiko that return the shell to the user.

Thanks in advance.

**Comment by ktbyers** [comment]
@jefpalma Netmiko does not have any interactive mode (I think that was what the question was asking).

**Comment by vimrichie** [comment]
I've done something like this before. It wasn't elegant but it worked for being 3 months into Python programming.

Tools used:
- Tmux
- Python 
- Netmiko (duh)

The process process
- Python would run a Tmux session detached. Detached meaning it would run a terminal session in the background.
- Then the code would proceed to use Netlike to do the automation work.
- Once work was completed it would print the output.
- Finally the Python script would notify me that in 10 seconds it was going to attach the screen in a new terminal tab. 

I would then see the new tab pop up and I would continue anything that I needed to do in the router. After some time I have modified my work flow and ended finding out that this process wasn't efficient. I noticed that Tmux has a command mode which might simplify this process. In the end I just scripted my terminal to SSH into the router and run my script in a new tab.
